Error packaging framework-only builds

<!-- TEMPLATE FOR BUG REPORTS -->

After building a framework-only build (see #17127), cpack reports errors attempting to produce a tgz file:

>CMake Error at /home/xmr/mantid-framework-test-build/cmake_install.cmake:63 (file):
>  file INSTALL cannot copy file
>  "/home/xmr/git/mantid/installers/LinuxInstaller/mantidplot.desktop" to
>  "/usr/share/applications/mantidplot.desktop".
>
>
>CMake Error at /home/xmr/mantid-framework-test-build/cmake_install.cmake:75 (file):
>  file INSTALL cannot copy file
>  "/home/xmr/git/mantid/images/MantidPlot_Icon_32offset.png" to
>  "/usr/share/pixmaps/mantidplot.png".

Since we didn't build MantidPlot, the two files don't exist.  It looks like the fix is to add another check in the top level CMakeLists.txt file around line 227.  This is probably limited to Linux (judging by the other tests in the CMakeLists.txt file).

To repeat the error:
```shell
cmake -G Ninja -DENABLE_MANTIDPLOT=FALSE -DENABLE_CPACK=TRUE ...
cmake --build . -- -j 20
cpack -G TGZ --config CPackConfig.cmake
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,7 +224,7 @@ endif ()
 install ( DIRECTORY installers/colormaps/ DESTINATION ${INBUNDLE}colormaps)
 
 # Install the files (.desktop and icon) to create a menu item, but only if installing to /opt/Mantid
-if ( ${CMAKE_SYSTEM_NAME} STREQUAL "Linux" AND CMAKE_INSTALL_PREFIX STREQUAL "/opt/Mantid" )
+if ( ENABLE_MANTIDPLOT AND ${CMAKE_SYSTEM_NAME} STREQUAL "Linux" AND CMAKE_INSTALL_PREFIX STREQUAL "/opt/Mantid" )
   install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/installers/LinuxInstaller/mantidplot.desktop
             DESTINATION /usr/share/applications )
   install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/images/MantidPlot_Icon_32offset.png

--- a/buildconfig/CMake/LinuxPackageScripts.cmake
+++ b/buildconfig/CMake/LinuxPackageScripts.cmake
@@ -135,12 +135,14 @@ endif ()
 
 # Local dev version
 set ( EXTRA_LDPATH "${ParaView_DIR}/lib" )
-set ( MANTIDPLOT_EXEC MantidPlot )
-configure_file ( ${CMAKE_MODULE_PATH}/Packaging/launch_mantidplot.sh.in
-                 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/launch_mantidplot.sh @ONLY )
-# Needs to be executable
-execute_process ( COMMAND "chmod" "+x" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/launch_mantidplot.sh"
-                  OUTPUT_QUIET ERROR_QUIET )
+if (ENABLE_MANTIDPLOT)
+  set ( MANTIDPLOT_EXEC MantidPlot )
+  configure_file ( ${CMAKE_MODULE_PATH}/Packaging/launch_mantidplot.sh.in
+                   ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/launch_mantidplot.sh @ONLY )
+  # Needs to be executable
+  execute_process ( COMMAND "chmod" "+x" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/launch_mantidplot.sh"
+                    OUTPUT_QUIET ERROR_QUIET )
+endif ()
 configure_file ( ${CMAKE_MODULE_PATH}/Packaging/mantidpython.in
                  ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/mantidpython @ONLY )
 # Needs to be executable
@@ -149,15 +151,17 @@ execute_process ( COMMAND "chmod" "+x" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/mantid
 
 # Package version
 set ( EXTRA_LDPATH "\${INSTALLDIR}/../lib/paraview-5.1" )
-set ( MANTIDPLOT_EXEC MantidPlot_exe )
-configure_file ( ${CMAKE_MODULE_PATH}/Packaging/launch_mantidplot.sh.in
-                 ${CMAKE_CURRENT_BINARY_DIR}/launch_mantidplot.sh.install @ONLY )
-install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/launch_mantidplot.sh.install
-          DESTINATION ${BIN_DIR} RENAME launch_mantidplot.sh
-          PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
-          GROUP_EXECUTE GROUP_READ
-          WORLD_EXECUTE WORLD_READ
-)
+if (ENABLE_MANTIDPLOT)
+  set ( MANTIDPLOT_EXEC MantidPlot_exe )
+  configure_file ( ${CMAKE_MODULE_PATH}/Packaging/launch_mantidplot.sh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/launch_mantidplot.sh.install @ONLY )
+  install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/launch_mantidplot.sh.install
+            DESTINATION ${BIN_DIR} RENAME launch_mantidplot.sh
+            PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
+            GROUP_EXECUTE GROUP_READ
+            WORLD_EXECUTE WORLD_READ
+  )
+endif ()
 configure_file ( ${CMAKE_MODULE_PATH}/Packaging/mantidpython.in
                  ${CMAKE_CURRENT_BINARY_DIR}/mantidpython.install @ONLY )
 install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/mantidpython.install

